Content/Bosses/GlassMiniboss/Projectiles.GlassBubble.cs

```diff
@@ -38,13 +45,37 @@ public override void SetDefaults()
 
 		public override void OnSpawn(IEntitySource source)
 		{
-			Helpers.Helper.PlayPitched("GlassMiniboss/WeavingSuper", 1f, 0f, Projectile.Center);
+			Projectile.scale = staticScaleToSet;
+			staticScaleToSet = 1f;
 		}
 
 		public override void AI()
 		{
+			if (!isLoaded)
+			{
+				Helpers.Helper.PlayPitched("GlassMiniboss/WeavingSuper", 1f, 0f, Projectile.Center);
+				isLoaded = true;
+			}
+
 			Timer++;
 
+			if (State == 0)
+			{
+				Glassweaver glassweaver = Parent.ModNPC as Glassweaver;
```

Comment:
we should null check this and have the projectile self-destruct incase the boss despawns